engine/src/object.cpp

```diff
@@ -317,11 +317,6 @@ MCObject::~MCObject()
 	MCMemoryDeleteArray(patterns);
 	deletepropsets();
 	MCValueRelease(tooltip);
-
-	// MW-2012-11-20: [[ IdCache ]] Make sure we delete the object from the cache - not
-	//   all deletions vector through 'scheduledelete'.
-	if (m_in_id_cache)
```

Comment:
I suggest putting MCAssert(!m_in_id_cache) here to ensure there aren't any odd cases :)